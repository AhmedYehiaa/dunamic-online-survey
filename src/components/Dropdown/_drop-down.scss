select {
  height: 34px;
  width: 40%;
  margin: 50px;
  padding: 5px 35px 5px 5px;
  border: 1px solid $dark-blue;
  font-size: 16px;
  background: $white;
}