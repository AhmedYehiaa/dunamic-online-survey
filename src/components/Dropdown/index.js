import React from 'react';
import PropTypes from 'prop-types';

const Dropdown = ({ options, selectValue, onChange }) => (
  <select onChange={onChange} value={selectValue}>
    {options.map(option => (
      <option key={option.value} value={option.value}>
        {option.name}
      </option>
    ))}
  </select>
);

Dropdown.propTypes = {
  options: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string.isRequired,
    value: PropTypes.string.isRequired,
  })).isRequired,
  selectValue: PropTypes.string,
  onChange: PropTypes.func.isRequired,
};

Dropdown.defaultProps = {
  selectValue: '',
};

export default Dropdown;
