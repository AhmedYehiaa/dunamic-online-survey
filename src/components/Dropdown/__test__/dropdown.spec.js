import React from 'react';
import { shallow } from 'enzyme';
import Dropdown from '../index';

it('it should renders the dropdown list', () => {
  const handleUpdatingAnswer = jest.fn();
  const options = [{
    name: 'name1',
    value: 'value1',
  }];
  const wrapper = shallow(
    <Dropdown
      options={options}
      value="value1"
      onChange={handleUpdatingAnswer}
    />,
  );
  expect(wrapper.children.length).toEqual(1);
  expect(wrapper).toMatchSnapshot();
});
