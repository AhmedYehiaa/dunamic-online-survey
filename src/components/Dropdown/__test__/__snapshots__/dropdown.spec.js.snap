// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should renders the dropdown list 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Dropdown
    onChange={[MockFunction]}
    options={
      Array [
        Object {
          "name": "name1",
          "value": "value1",
        },
      ]
    }
    value="value1"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <option
          value=""
        >
          Please choose an option
        </option>,
        Array [
          <option
            value="value1"
          >
            name1
          </option>,
        ],
      ],
      "onChange": [MockFunction],
      "value": "value1",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Please choose an option",
          "value": "",
        },
        "ref": null,
        "rendered": "Please choose an option",
        "type": "option",
      },
      Object {
        "instance": null,
        "key": "value1",
        "nodeType": "host",
        "props": Object {
          "children": "name1",
          "value": "value1",
        },
        "ref": null,
        "rendered": "name1",
        "type": "option",
      },
    ],
    "type": "select",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <option
            value=""
          >
            Please choose an option
          </option>,
          Array [
            <option
              value="value1"
            >
              name1
            </option>,
          ],
        ],
        "onChange": [MockFunction],
        "value": "value1",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Please choose an option",
            "value": "",
          },
          "ref": null,
          "rendered": "Please choose an option",
          "type": "option",
        },
        Object {
          "instance": null,
          "key": "value1",
          "nodeType": "host",
          "props": Object {
            "children": "name1",
            "value": "value1",
          },
          "ref": null,
          "rendered": "name1",
          "type": "option",
        },
      ],
      "type": "select",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
